### PR TITLE
Cut 2.5.0-rc.0

### PR DESCRIPTION
**Pull request by brancz**
Pretty impressive list for a 6 week time window. Please check and make sure the entries I added make sense and that all work is reflected :slightly_smiling_face: .

@brian-brazil @juliusv @krasi-georgiev @simonpasquier @gouthamve @cstyan @csmarchbanks

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,36 @@
+## 2.5.0-rc.0 / 2018-10-23
+
+Starting with the s390x platform for Linux is officially supported.
```
Comment (mdlayher) [inline CHANGELOG.md]: * "Starting with this release"?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+
+* [CHANGE] Remote write default for a shards buffer capacity was changed from `100000` to `10000`.
```
Comment (gouthamve) [inline CHANGELOG.md]: This made it into `2.4.0`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+* [CHANGE] Remote write default for a retries was changed from `10` to `3`.
```
Comment (gouthamve) [inline CHANGELOG.md]: Same for this.

File: CHANGELOG.md (hunk 1 above)
Comment (SuperQ) [reply CHANGELOG.md]: We tend to not list changes like build stuff in our release notes.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+* [ENHANCEMENT] Remote Read: Add `prometheus_api_remote_read_queries` metric tracking currently executed or waiting remote read queries.
+* [ENHANCEMENT] Remote Read: Reduced memory usage.
+* [ENHANCEMENT] Discovery: Split `prometheus_sd_discovered_targets` metric by scrape and notify as well as by section in the respective configuration.
+* [ENHANCEMENT] Discovery: Add `prometheus_sd_discovered_targets`, `prometheus_sd_received_updates_total`, `prometheus_sd_updates_delayed_total`, and `prometheus_sd_updates_total` metrics for discovery subsystem.
+* [ENHANCEMENT] Discovery: Improve performance of previously slow updates of changes of targets.
+* [ENHANCEMENT] Kubernetes SD: Add extended metrics.
+* [ENHANCEMENT] OpenStack SD: Support discovering instances from all projects.
+* [ENHANCEMENT] OpenStack SD: Discover all interfaces.
+* [ENHANCEMENT] OpenStack SD: Support `tls_config` for the used HTTP client.
+* [ENHANCEMENT] EC2 SD: Expose owner of an instance as meta label.
```
Comment (gouthamve) [inline CHANGELOG.md]: Made it into 2.4.0

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,36 @@
...
+* [ENHANCEMENT] Triton SD: Add ability to filter triton_sd targets by pre-defined groups.
+* [ENHANCEMENT] Web UI: Avoid browser spell-checking in expression field.
+* [ENHANCEMENT] Web UI: Add scrape duration and last evaluation time in targets and rules pages.
+* [ENHANCEMENT] Web UI: Improve rule view by wrapping lines.
+* [BUGFIX] Fix histograms on 32 bit platforms.
```
Comment (gouthamve) [inline CHANGELOG.md]: This never made it into a release and can be removed

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,36 @@
...
+
+* [CHANGE] Remote write default for a shards buffer capacity was changed from `100000` to `10000`.
+* [CHANGE] Remote write default for a retries was changed from `10` to `3`.
+* [FEATURE] Add OpenMetrics support (EXPERIMENTAL).
+* [FEATURE] Add unit testing for rules.
+* [ENHANCEMENT] Make maximum number of samples per query configurable via `--query.max-samples` flag.
+* [ENHANCEMENT] Make maximum number of concurrent remote reads configurable via `--storage.remote.read-concurrent-limit` flag.
+* [ENHANCEMENT] Remote Read: Add `prometheus_api_remote_read_queries` metric tracking currently executed or waiting remote read queries.
+* [ENHANCEMENT] Remote Read: Reduced memory usage.
+* [ENHANCEMENT] Discovery: Split `prometheus_sd_discovered_targets` metric by scrape and notify as well as by section in the respective configuration.
```
Comment (brian-brazil) [inline CHANGELOG.md]: This is a CHANGE

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,36 @@
...
+* [ENHANCEMENT] OpenStack SD: Support `tls_config` for the used HTTP client.
+* [ENHANCEMENT] EC2 SD: Expose owner of an instance as meta label.
+* [ENHANCEMENT] Triton SD: Add ability to filter triton_sd targets by pre-defined groups.
+* [ENHANCEMENT] Web UI: Avoid browser spell-checking in expression field.
+* [ENHANCEMENT] Web UI: Add scrape duration and last evaluation time in targets and rules pages.
+* [ENHANCEMENT] Web UI: Improve rule view by wrapping lines.
+* [BUGFIX] Fix histograms on 32 bit platforms.
+* [BUGFIX] Change max/min over_time to handle NaNs properly.
+* [BUGFIX] Discovery: Properly unregister source when targets are empty.
+* [BUGFIX] Fix asset paths for Windows platforms.
```
Comment (simonpasquier) [inline CHANGELOG.md]: Was already released in v2.4.1

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,36 @@
...
+* [BUGFIX] Don't remove final dot from rooted DNS names.
```
Comment (simonpasquier) [inline CHANGELOG.md]: This one is still controversial, see #4766 .

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,36 @@
+## 2.5.0-rc.0 / 2018-10-23
+
+Starting with the s390x platform for Linux is officially supported.
+
+* [CHANGE] Remote write default for a shards buffer capacity was changed from `100000` to `10000`.
+* [CHANGE] Remote write default for a retries was changed from `10` to `3`.
+* [FEATURE] Add OpenMetrics support (EXPERIMENTAL).
```
Comment (brian-brazil) [inline CHANGELOG.md]: For scraping.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,36 @@
...
+* [ENHANCEMENT] OpenStack SD: Discover all interfaces.
+* [ENHANCEMENT] OpenStack SD: Support `tls_config` for the used HTTP client.
+* [ENHANCEMENT] EC2 SD: Expose owner of an instance as meta label.
+* [ENHANCEMENT] Triton SD: Add ability to filter triton_sd targets by pre-defined groups.
+* [ENHANCEMENT] Web UI: Avoid browser spell-checking in expression field.
+* [ENHANCEMENT] Web UI: Add scrape duration and last evaluation time in targets and rules pages.
+* [ENHANCEMENT] Web UI: Improve rule view by wrapping lines.
+* [BUGFIX] Fix histograms on 32 bit platforms.
+* [BUGFIX] Change max/min over_time to handle NaNs properly.
+* [BUGFIX] Discovery: Properly unregister source when targets are empty.
```
Comment (simonpasquier) [inline CHANGELOG.md]: This bug fix was in the file SD adapter code so it probably doesn't warrant an entry.

**Comment by brancz** [comment]
@gouthamve PR numbering we only started doing recently but happy to add it, will make sure to add it to https://github.com/prometheus/docs/pull/1169 as well then.

File: CHANGELOG.md (hunk 1 above)
Comment (brancz) [reply CHANGELOG.md]: @SuperQ agreed, but supporting an additional platform seems like it belongs in the changelog. I'm actually interested what we mean by this (I took the "support" statement right from the PR that merged this). For this and other platforms that are not amd64 we should probably clearer on what support means as we only run tests on amd64 and arguably all of us probably don't run Prometheus on anything other (maybe some arm devices). So "supportability" becomes somewhat undefined.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,36 @@
+## 2.5.0-rc.0 / 2018-10-23
+
+Starting with the s390x platform for Linux is officially supported.
+
+* [CHANGE] Remote write default for a shards buffer capacity was changed from `100000` to `10000`.
+* [CHANGE] Remote write default for a retries was changed from `10` to `3`.
+* [FEATURE] Add OpenMetrics support (EXPERIMENTAL).
+* [FEATURE] Add unit testing for rules.
+* [ENHANCEMENT] Make maximum number of samples per query configurable via `--query.max-samples` flag.
```
[the remark this answers is not in the document]
Comment (cstyan) [reply CHANGELOG.md]: :+1: 

(pr #4513 )

File: CHANGELOG.md (hunk 8 above)
Comment (grobie) [reply CHANGELOG.md]: I'm in favor of reverting this change, see my comment in #4766. If you decide against it, this is a significant `CHANGE` for any dns discovery user and should be highlighted as such.

**Comment by brancz** [comment]
I updated the PR to inline the respective pull request numbers and updated the release date to today's date. It seems we also want to have some consensus on https://github.com/prometheus/prometheus/pull/4766.

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,32 @@
+## 2.5.0-rc.0 / 2018-10-24
+
+* [CHANGE] Marathon SD: Various changes to adapt to marathon 1.5+. #4499
```
Comment (juliusv) [inline CHANGELOG.md]:
```suggestion
* [CHANGE] Marathon SD: Various changes to adapt to Marathon 1.5+. #4499
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,32 @@
...
+* [CHANGE] Discovery: Split `prometheus_sd_discovered_targets` metric by scrape and notify as well as by section in the respective configuration. #4753
```
Comment (juliusv) [inline CHANGELOG.md]:
```suggestion
* [CHANGE] Discovery: Split `prometheus_sd_discovered_targets` metric by scrape and notify (Alertmanager SD) as well as by section in the respective configuration. #4753
```

**Comment by juliusv** [comment]
At least from a breakage perspective (as outlined by @grobie this can be quite severe) I'd be for #4766, that is, reverting the DNS dot removal fix and going back to the behavior we had.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**